File: src/day6/haskell/part2.hs
import Data.List.Split
import Data.List

main = do
  lines <- readFile "input"
  let groups = splitOn "\n\n" lines
  let sgroups = map removeEmptyElems $ map (splitOn "\n") groups
  let commons = map getCommon sgroups
  let ans = sum $ map length commons
  putStrLn $ show ans

removeEmptyElems [] = []
removeEmptyElems ("":s) = removeEmptyElems s
removeEmptyElems (x:s) = x: removeEmptyElems s

getCommon :: [String] -> String
getCommon s = nub [c | x<-s, c<-x, isInAll c s]

isInAll :: Char -> [String] -> Bool
isInAll c s = all (==True) [c `elem` x | x<-s]

File: src/day6/python/part2.py
f = open("input", "r")
lines = f.read().split("\n\n")

parsed = [x.split("\n") for x in lines]

noempty = [[x for x in l if x != ''] for l in parsed]

def getCommon(arr):
    head, *rest = arr
    ret = head
    for r in rest:
        ret = list(set(ret).intersection(set(r)))
    return ret

uniq = [len(getCommon(x)) for x in noempty]

print(sum(uniq))
